integration-tests/src/tests/client/state_snapshot.rs

```diff
@@ -243,7 +244,8 @@ fn test_make_state_snapshot() {
         )
     );
 
-    // check that if the snapshot is deleted from file system while there's entry in DBCol::STATE_SNAPSHOT_KEY and write lock is nonempty, making a snpashot of the same hash will not write to the file system
```

Comment:
Why would we have this previous behavior?? I would have expected calling a new create snapshot request would have.. well.. created a snapshot